Source/BoundaryConditions/PML.H

```diff
@@ -112,7 +112,7 @@ public:
     }
 
 private:
-    const amrex::BoxArray& m_grids;
```

Comment:
I am not fully convinced about this substitution. Maybe a (possibly smart) pointer would be better, in order to avoid copies.

Reply:
Hm, also should this change when we load balance?

cc @WeiqunZhang

Reply:
I think we can change it to `const amrex::BoxArray*` and we should not make a copy. The cost of copy itself is actually not that expensive because the data inside BoxArray are stored with shared_ptr. But we only need to use the BoxArray when constructing PML. After a PML object is constructed, we no long need that BoxArray. So there is no reason to hold a copy of it.

Reply:
Thanks, @WeiqunZhang . I've followed your suggestion.